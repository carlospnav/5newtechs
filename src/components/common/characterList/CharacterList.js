import styled from "styled-components"
import Avatar from "common/avatar/Avatar"
import Link from "common/Link"
import { flexGroup } from "theme/utils"
import css from "theme/variables"
import map from "lodash/map"

const Item = styled.li`
  ${flexGroup("column", "center", "center")};
  border: 1px solid ${css.color.borderColor};
  border-radius: 6px;
  height: 260px;
  width: 220px;
  margin: 0 10px 20px 10px;
`

const List = styled.ul`
  ${props => props.center ? flexGroup(null, "center") : flexGroup()};
  flex-flow: row wrap;
  padding: 0;
  margin: 0;
`
const Name = styled.h2`
  color: #a0a9ab;
  font-size: 18px;
  font-weight: ${css.fontWeight.regular};
  font-family: ${css.font.roboto};
  margin: 30px 0 0 0;
  &:hover{ 
    font-weight: ${css.fontWeight.bold};
    cursor: pointer;
   }
`

export default function({items, center}){
  return(
    <List center={center}>
      {map(items, item => (
              <Item key={item.id}>
                <Avatar small/>
                <Link route="person_detail" personId={item.id}>
                  <Name>{item.name}</Name>
                </Link>
              </Item>
            ))}
    </List>
  )
}