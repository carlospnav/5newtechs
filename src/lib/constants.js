export const fetchPolicyList = {
  fetchPolicy: "network-only"
}
