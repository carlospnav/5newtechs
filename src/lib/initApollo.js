import { ApolloClient } from "apollo-client"
import { HttpLink } from "apollo-link-http"
import { InMemoryCache } from "apollo-cache-inmemory"
import fetch from "isomorphic-fetch"
import gql from "graphql-tag"
let apolloClient = null

if (!process.browser) {
  global.fetch = fetch
}

function create({ state }) {
  const httpLink = new HttpLink({
    ssrMode: !process.browser,
    uri: process.env.GRAPHQL_API_URL
  })

  return new ApolloClient({
    link: httpLink,
    cache: new InMemoryCache().restore(state)
  })
}

export default function initApollo(options) {
  if (!process.browser) {
    return create(options)
  }

  // Reuse client on the client-side
  if (!apolloClient) {
    apolloClient = create(options)
  }

  return apolloClient
}
