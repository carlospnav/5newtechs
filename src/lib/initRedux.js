import { createStore, combineReducers, compose } from "redux"
import url from "./reducers/url"

let reduxStore = null

// Get the Redux DevTools extension and fallback to a no-op function
let devtools = f => f
if (process.browser && window.__REDUX_DEVTOOLS_EXTENSION__) {
  devtools = window.__REDUX_DEVTOOLS_EXTENSION__()
}

function create() {
  return createStore(
    combineReducers({
      // Setup reducers
      url
    }),
    compose(devtools)
  )
}

export default function initRedux() {
  if (!process.browser) {
    return create()
  }

  if (!reduxStore) {
    reduxStore = create()
  }

  return reduxStore
}
