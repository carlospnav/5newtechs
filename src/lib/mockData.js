export const films = [
  {
    id: "1",
    episodeID: 1,
    title: "A New Hope",
    releaseDate: "2010"
  },
  {
    id: "2",
    episodeID: 2,
    title: "A First Hope",
    releaseDate: "2011"
  },
  {
    id: "3",
    episodeID: 3,
    title: "A Last Hope",
    releaseDate: "2012"
  }
]

export const film = {
  id: "1",
  episodeID: 1,
  title: "A First Hope Returns Wars",
  release: "2009",
  content: "Damn you, openingCrawl!",
  director: "Michael Bay"
}

export const person = {
  personId:"1",
  name: "Tauntaun",
  homeworld: {
    name: "Hoth"
  },
  gender: "Male",
  birthYear: "1001",
  skinColor: "White",
  hairColor: "White",
  eyeColor: "Black"
}

export const people = [
  {
    personId:"1",
    name: "Tauntaun",
    homeworld: {
      name: "Hoth"
    },
    gender: "Male",
    birthYear: "1001",
    skinColor: "White",
    hairColor: "White",
    eyeColor: "Black"
  },
  {
    personId:"2",
    name: "Stormtrooper's stray blaster shot",
    homeworld: {
      name: "Imperial Blaster"
    },
    gender: "Neutral",
    birthYear: "0000",
    skinColor: "Translucent",
    hairColor: "Translucent",
    eyeColor: "Redish"
  }
]

