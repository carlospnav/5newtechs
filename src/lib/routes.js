const routes = (module.exports = require("next-routes")())

routes.add("people").add("person_detail", "/people/:personId").add("movies", "/movies/:filmId")
