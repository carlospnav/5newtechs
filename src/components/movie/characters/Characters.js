import styled from "styled-components"
import css from "theme/variables"
import { flexGroup } from "theme/utils"
import CharacterList from "common/characterList/CharacterList"

const Characters =  styled.section`
  ${flexGroup()};
  flex-wrap: wrap;
  width: 60%;

  @media (min-width: 1000px){
    width: 100%;
  }
`

const Title = styled.h2`
  width: 100%;
  font-family: ${css.font.roboto};
  font-size: 22px;
  font-weight: ${css.fontWeight.semibold};
  letter-spacing: 0.8px;
  color: ${css.color.blueTitle};
  margin: 40px 10px;
`

export default function({items}){
  return (
    <Characters>
      <Title>Characters</Title>
      <CharacterList items={items}/>
    </Characters>
  )
}