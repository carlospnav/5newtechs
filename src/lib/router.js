import { Router as NextRouter } from "lib/routes"

export default function Router({ returnTo }) {
  return {
    goToPeople() {
      return NextRouter.pushRoute("people")
    },
    goBack(orCallThis) {
      if (returnTo) {
        return () => NextRouter.pushRoute(returnTo)
      } else {
        return orCallThis
      }
    }
  }
}
