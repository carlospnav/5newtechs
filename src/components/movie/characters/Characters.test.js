import { shallow } from "enzyme"
import Characters, { Title } from "./Characters"
import CharacterList from "common/characterList/CharacterList"
import { people } from "lib/mockData"
import "jest-styled-components"

describe("<PersonDetail>", () => {
  let wrapper;

  function personDetail(){
    wrapper = shallow(<Characters items={people} />)
  }

  beforeEach(personDetail)
  afterEach(() => wrapper = undefined)

  it("renders a characters list properly.", () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })

  it("renders the Title component properly.", () => {
    expect(wrapper.find(Title).dive()).toMatchSnapshot()
  })

  it("renders a CharacterList with people as its props.", () => {
    expect(wrapper.find(CharacterList).prop('items')).toEqual(people)
  })
})