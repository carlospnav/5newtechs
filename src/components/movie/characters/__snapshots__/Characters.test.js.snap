// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonDetail> renders a characters list properly. 1`] = `
.c0 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 60%;
}

@media (min-width:1000px) {
  .c0 {
    width: 100%;
  }
}

<section
  className="c0"
>
  <styled.h2>
    Characters
  </styled.h2>
  <Component
    items={
      Array [
        Object {
          "birthYear": "1001",
          "eyeColor": "Black",
          "gender": "Male",
          "hairColor": "White",
          "homeworld": Object {
            "name": "Hoth",
          },
          "name": "Tauntaun",
          "personId": "1",
          "skinColor": "White",
        },
        Object {
          "birthYear": "0000",
          "eyeColor": "Redish",
          "gender": "Neutral",
          "hairColor": "Translucent",
          "homeworld": Object {
            "name": "Imperial Blaster",
          },
          "name": "Stormtrooper's stray blaster shot",
          "personId": "2",
          "skinColor": "Translucent",
        },
      ]
    }
  />
</section>
`;

exports[`<PersonDetail> renders the Title component properly. 1`] = `
.c0 {
  width: 100%;
  font-family: 'Roboto',sans-serif;
  font-size: 22px;
  font-weight: 600;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
  color: #9fc7ef;
  margin: 40px 10px;
}

<h2
  className="c0"
>
  Characters
</h2>
`;
